fix(pipelined): Fix Unit Test arg init for qos, add ebpf guards

koolzz opened the pull request:
Signed-off-by: Nick Yurchenko <koolzz@fb.com>

<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary
 -  Qos init seems to fail in unit tests because in qos init we don't initiliaze all the args because of the premature return statements (introduced here https://github.com/magma/magma/pull/11287, cc @ankitbhadoria21 )
 - Also noticed some ebpf issues with nat_iface lookup in tests, added some checcks

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

koolzz committed:
fix(pipelined): Fix Unit Test arg init for qos, add ebpf guards

Signed-off-by: Nick Yurchenko <koolzz@fb.com>

themarwhal commented:
force merging to unblock lte-integ-test on CI

ankitMeta1 commented:
I am actually curious about this one @koolzz, looking at the changes you made here and I made in my mentioned PR, it looks like the the check I added to see whether 'qos' key exists in config or not, in the QoSManager constructor caused it?, I added it because we were accessing the 'qos' section from the config, and if it doesn't exist, then an exception will be thrown, so we should probably check for it, before accessing.

The old code operates like this, it tries to access 'qos' field and call setup, but if an exception is thrown and we will not call setup in QoSManager. So I am just wondering if previously the test was previously silently not running?

koolzz commented:
> I am actually curious about this one @koolzz, looking at the changes you made here and I made in my mentioned PR, it looks like the the check I added to see whether 'qos' key exists in config or not, in the QoSManager constructor caused it?, I added it because we were accessing the 'qos' section from the config, and if it doesn't exist, then an exception will be thrown, so we should probably check for it, before accessing.
> 
> The old code operates like this, it tries to access 'qos' field and call setup, but if an exception is thrown and we will not call setup in QoSManager. So I am just wondering if previously the test was previously silently not running?

Yeah the error was in the  init_impl function as the variable that we were checki was not initialized

ankitMeta1 commented:
> > I am actually curious about this one @koolzz, looking at the changes you made here and I made in my mentioned PR, it looks like the the check I added to see whether 'qos' key exists in config or not, in the QoSManager constructor caused it?, I added it because we were accessing the 'qos' section from the config, and if it doesn't exist, then an exception will be thrown, so we should probably check for it, before accessing.
> > The old code operates like this, it tries to access 'qos' field and call setup, but if an exception is thrown and we will not call setup in QoSManager. So I am just wondering if previously the test was previously silently not running?
> 
> Yeah the error was in the init_impl function as the variable that we were checki was not initialized

Cool. then I think my other hypothesis is also right. becuase I didn't reorder the initialization of variables in  the constructor, I just added a condition to check for 'qos'. So in the previous code before my change, if you were to try to access 'qos' from config, it will throw exception(https://github.com/magma/magma/pull/11287/files#diff-afaf023d5c72e158db82116e628ecf30043e30cb73d8482163e56150eef37c6fL244) and then we won't call setup(https://github.com/magma/magma/pull/11287/files#diff-afaf023d5c72e158db82116e628ecf30043e30cb73d8482163e56150eef37c6fL183-L184) or initialize QosStore. So I don't think those tests were running correctly before? maybe they were silently off? @koolzz @pshelar
